src/toil/utils/toilLaunchCluster.py

```diff
@@ -131,14 +166,35 @@ def main():
                           awsEc2ProfileArn=options.awsEc2ProfileArn,
                           awsEc2ExtraSecurityGroupIds=options.awsEc2ExtraSecurityGroupIds)
 
-    for worker_node_type, num_workers in zip(worker_node_types, worker_quantities):
-        if ':' in worker_node_type:
-            worker_node_type, bid = worker_node_type.split(':', 1)
-            cluster.addNodes(nodeType=worker_node_type,
-                             numNodes=int(num_workers),
-                             preemptable=True,
-                             spotBid=float(bid))
+    for typeNum, count in enumerate(fixedNodeCounts):
+        # For each batch of workers to make at startup
+        if count == 0:
+            # Don't want any
+            continue
+        wanted = parsedNodeTypes[typeNum]
+        if wanted[1] is None:
+            # Make non-spot instances
+            cluster.addNodes(nodeType=wanted[0], numNodes=count, preemptable=False)
+        else:
+            # We have a spot bid
+            cluster.addNodes(nodeType=wanted[0], numNodes=count, preemptable=True,
+                             spotBid=wanted[1])
+                             
+    for typeNum, count in enumerate(managedNodeCounts):
+        # For each batch of workers to dynamically scale
+        if count == 0:
+            # Don't want any
+            continue
+        wanted = parsedNodeTypes[typeNum]
+        if wanted[1] is None:
+            # Make non-spot instances
+            cluster.addManagedNodes(nodeType=wanted[0], maxNodes=count, preemptable=False)
         else:
-            cluster.addNodes(nodeType=worker_node_type,
-                             numNodes=int(num_workers),
-                             preemptable=False)
+            # Bid at the given price.
+            cluster.addManagedNodes(nodeType=wanted[0], maxNodes=count, preemptable=True,
+                                    spotBid=wanted[1])
```

Comment:
These two loops are very similar, and it might be worth consolidating them as one function.  Not sure it's a big gain though, so not a strong suggestion.

Reply:
I've put these together into one loop. I'm also not sure it's better.